### DISCUSSION TITLE
How to render validation ErrorMessage in the UI instead of a popup in Next.js?

### DISCUSSION THREAD

**Discussion by arsekil** [colinhacks/zod#3013] [Q&A] [answered]
Hello everyone,

My question is that i want to bypass the way Next.js shows errors and I want to render the parse error messages in the UI with styling.

can i override the way these validation errors show up with teh ZodErrorMap?

**Comment by JacobWeisenburger** [comment, marked as answer]
Perhaps this will help you:
https://github.com/colinhacks/zod/blob/master/ERROR_HANDLING.md#customizing-errors-with-zoderrormap

If you found my answer satisfactory, please consider supporting me. Even a small amount is greatly appreciated. Thanks friend! 🙏
https://github.com/sponsors/JacobWeisenburger

**Comment by JacobWeisenburger** [comment]
If you provide a code example that is causing problems for you, it would be easier to help you.
